go/enclave/rpc/GetTransactionReceipt.go

```diff
@@ -57,3 +76,108 @@ func GetTransactionReceiptExecute(builder *CallBuilder[gethcommon.Hash, map[stri
 	builder.ReturnValue = &r
 	return nil
 }
+
+func fetchFromCache(ctx context.Context, storage storage.Storage, cacheService *storage.CacheService, txHash gethcommon.Hash, requester *gethcommon.Address) (map[string]interface{}, error) {
+	rec, _ := cacheService.ReadReceipt(ctx, txHash)
+	if rec == nil {
+		return nil, nil
+	}
+
+	// receipt found in cache
+	// for simplicity only the tx sender will access the cache
+	// check whether the requester is the sender
+	if rec.From != requester {
+		return nil, nil
+	}
+
+	logs := rec.Receipt.Logs
+	// filter out the logs that the sender can't read
+	// doesn't apply to value transfers (when to=nil)
```

Comment:
Value transfers have a To address I think, it's only contract creation where To=nil isn't it? Is it ok that this returns error for value transfers?